### PR TITLE
[DebugInfo] Enable deprecation of iterator-insertion methods

### PR DESCRIPTION
**Pull request by jmorse**
Propagate to some other utilities that shouldn't be used, but we don't want to delete just yet. There's some final cleanup in the LLVM Examples which usually don't get built: otherwise everything in the monorepo seems to build cleanly.

I've also added a cast to the default-nullptr PHINode::Create method. Its default-form seems pretty widespread, doing this causes most uses to take the "Being inserted in a BasicBlock that happens to be absent" path as opposed to the "Being inserted with an instruction pointer, which is deprecated" path. We should still get deprecation warnings if anyone passes in an instruction pointer directly.

### CODE REVIEW

File: llvm/include/llvm/IR/Instructions.h [hunk 1]
```diff
@@ -2537,7 +2537,7 @@ class PHINode : public Instruction {
   /// edges that this phi node will have (use 0 if you really have no idea).
   static PHINode *Create(Type *Ty, unsigned NumReservedValues,
                          const Twine &NameStr = "",
-                         InsertPosition InsertBefore = nullptr) {
+                         InsertPosition InsertBefore = (BasicBlock*)nullptr) {
```
Comment (nikic) [inline llvm/include/llvm/IR/Instructions.h:2540]: Why is this necessary despite the `nullptr_t` overload on InsertPosition?

Comment (jmorse) [reply llvm/include/llvm/IR/Instructions.h]: Apparently it isn't (now removed). Not sure what I was doing before, but I thought I was getting a bunch of deprecated warnings out of here, which I now can't replicate.

File: llvm/include/llvm/IR/InstrTypes.h [hunk 2]
```diff
@@ -60,6 +60,8 @@ class UnaryInstruction : public Instruction {
       : Instruction(Ty, iType, &Op<0>(), 1, IB) {
     Op<0>() = V;
   }
+  LLVM_DEPRECATED("Use BasicBlock::iterators for insertion instead",
+                  "BasicBlock::iterator")
```
Comment (SLTozer) [inline llvm/include/llvm/IR/InstrTypes.h:64]: I think these (and the other cases in this file where you've added `LLVM_DEPRECATED`) could be replaced with `InsertPosition` arguments - especially the `HANDLE_` cases, where we have 3 different methods for each type, which all forward to the same `InsertPosition` fn underneath. Not sure why I missed these the first time, but (as long as it works) it should probably be done now.

**Review by SLTozer — COMMENTED** [review]
Changes largely LGTM, but an inline comment about maybe changing some function signatures.

**Comment by jmorse** [comment]
Thanks for the insert-position tips, I've consolidated all of those into InsertPosition object arguments.

**Review by SLTozer — APPROVED** [review]
